nsqlookupd/registration_db.go

```diff
@@ -30,10 +30,16 @@ type PeerInfo struct {
 	Version          string `json:"version"`
 }
 
+type State struct {
+	Paused bool `json:"paused"`
+}
+
 type Producer struct {
 	peerInfo     *PeerInfo
 	tombstoned   bool
 	tombstonedAt time.Time
+	topicState   *State
+	channelState *State
```

Comment:
Let's just add a field `paused` here, it's internal.

Also these don't need to be pointer types.